File: Linked list/Reverse Linked List/Reverse Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        stack<int> reverse;
        if(head == NULL) return head;
        ListNode* ptr = head;
        ListNode* rev = new ListNode(0);
        ListNode* point = rev;
        while(ptr != NULL){
           reverse.push(ptr->val);
           ptr = ptr->next;
        }
        while(!reverse.empty()){
            int f = reverse.top();
            rev->next = new ListNode(f); 
            rev = rev->next;
            reverse.pop();
        }
     return point->next;
    }
};

File: Linked list/Reverse Linked List/Reverse Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if(head == null) return head;
        ListNode rev = new ListNode(0);
        ListNode ptr = head;
        ListNode point = rev;
        Stack <Integer> st = new Stack<>();
        while(ptr != null){
            st.push(ptr.val);
            ptr = ptr.next;
        }
        while(!st.isEmpty()){
            int f = st.peek();
            rev.next = new ListNode(f);
            rev = rev.next;
            st.pop();
        }
        return point.next;
    }
}
